api/buckets/service.go

```diff
@@ -2131,18 +2214,46 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 				return nil, fmt.Errorf("there is an in progress archive")
 			// Case 1.c.
 			case ffs.Failed, ffs.Canceled:
-				jid, err = s.PGClient.FFS.PushStorageConfig(ctxFFS, p.Cid(), powc.WithOverride(true))
+				jid, err = s.PGClient.FFS.PushStorageConfig(ctxFFS, p.Cid(), powc.WithStorageConfig(storageConfig), powc.WithOverride(true))
 				if err != nil {
 					return nil, fmt.Errorf("pushing config: %s", err)
 				}
 			default:
 				return nil, fmt.Errorf("unexpected current archive status: %d", ba.Archives.Current.JobStatus)
 			}
 		} else { // Case 2.
-			jid, err = s.PGClient.FFS.Replace(ctxFFS, oldCid, p.Cid())
+			res, err := s.PGClient.FFS.GetStorageConfig(ctxFFS, oldCid)
 			if err != nil {
-				return nil, fmt.Errorf("replacing cid: %s", err)
+				return nil, fmt.Errorf("looking up old storage config: %s", err)
+			}
+			// ToDo: Remove this once .PGClient.FFS.GetStorageConfig returns a ffs.StorageConfig
+			var oldStorageConfig *ffs.StorageConfig
+			if res.Config != nil {
+				oldStorageConfig = &ffs.StorageConfig{
+					Repairable: res.Config.Repairable,
+					Cold:       fromRPCColdConfig(res.Config.Cold),
+					Hot:        fromRPCHotConfig(res.Config.Hot),
+				}
 			}
+
+			if reflect.DeepEqual(&storageConfig, oldStorageConfig) {
```

Comment:
Here's that comparison. Cool to use `reflect`?

Reply:
I had read that `cmp.Equal` is meant for testing and panics under some circumstances. Not sure if that is true, didn't dig much deeper.

Reply:
`cmp.Equal` was created as a better solution than `reflect.DeepEqual` which compares unexported fields, and might not behave correctly on particular types which its equality might not be easily comparable by bytes

I'd say both are meant for a quick way of comparing things. The full way is manual.

Reply:
Cool, switched to `cmd.Equal`